### PR TITLE
feat(AutoLayout): add IsReverse Property

### PR DESCRIPTION
**Pull request by Robert-Louis**
GitHub Issue (If applicable): #
https://github.com/unoplatform/Uno.Figma/issues/550

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

- Children have the Z order according to the order they are added

## What is the new behavior?

- Add IsReverse  property to AutoLayout that reverse the Z order
This property emulate the Canvas Stacking option for AutoLayout  in figma 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/controls/AutoLayoutControl.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ Spacing | double | Gets or sets the uniform distance (in pixels) between grid. E
 Justify | AutoLayoutJustify | Either Stack, space between children behave like a `StackPanel`, or SpaceBetween, between each children a equal space is added assuring that all the `AutoLayout` space is occupy. Note : if a child has his `PrimaryAlignment` set to Stretch, it will default to Stack.
 Children |  AutoLayoutChildren | List of FrameworkElement that are the AutoLayout children
 PrimaryAxisAlignment | AutoLayoutAlignment | Indicates where the content be displayed on the axis of the AutoLayout orientation
+IsReverse | bool | Reverse the Z order of the children
```
Comment (kazo0) [inline doc/controls/AutoLayoutControl.md:38]:
```suggestion
IsReverse | bool | Reverse the Z-index of the children
```

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs [hunk 2]
```diff
@@ -34,6 +34,15 @@ protected override void OnApplyTemplate()
...
 
+		public static readonly DependencyProperty IsReverseProperty = DependencyProperty.Register(
+			"IsReverse", typeof(bool), typeof(AutoLayout), new PropertyMetadata(default(bool), propertyChangedCallback: UpdateCallback));
+
+		public bool IsReverse
+		{
+			get => (bool)GetValue(IsReverseProperty);
+			set => SetValue(IsReverseProperty, value);         
+		}
+
```
Comment (kazo0) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs:45]: Is the use of a single `IsReverse` property just something that will make it easier for code generation? Wouldn't it be better to implement something like a `AutoLayout.ZIndex` attached property similar to the `Canvas.ZIndex`?

Comment (carldebilly) [reply src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs:45]: ZIndexes would be a lot more complicated to implement and the actual current requirement is just a way to reverse the stacking order of children.

Comment (Robert-Louis) [reply src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs:45]: It's a different use of Canvas.ZIndex. It's not for the ZIndex of each item but to reverse the order of the all children. 
Should I call it IsReverseZIndex for more lisibility ?

**Pull request merged**